test-common/eos-test-utils.c

```diff
@@ -580,7 +580,15 @@ prepare_commit (GFile *repo,
     g_autoptr(GFile) commit_file = g_file_get_child (tree_root, commit_filename);
 
     if (g_file_query_exists (commit_file, NULL))
-      return TRUE;
+      {
+        if (out_checksum != NULL)
+          return get_current_commit_checksum (repo,
+                                              collection_ref,
+                                              out_checksum,
+                                              error);
+
+        return TRUE;
```

Comment:
This chunk looks like a separate change to fix the commit-already-exists path in `prepare_commit()` to correctly set `out_checksum`. It should be split out into a separate commit, and the commit message should briefly mention what it’s fixing.